### PR TITLE
 ptr as boolean

### PR DESCRIPTION
**Pull request by 1bsyl**
some attemps for https://github.com/libsdl-org/SDL/issues/7214

### CODE REVIEW

File: src/SDL_assert.c [hunk 1]
```diff
@@ -106,11 +106,11 @@ static void SDL_GenerateAssertionReport(void)
     const SDL_AssertData *item = triggered_assertions;
 
     /* only do this if the app hasn't assigned an assertion handler. */
-    if ((item != NULL) && (assertion_handler != SDL_PromptAssertion)) {
+    if ((item) && (assertion_handler != SDL_PromptAssertion)) {
```
Comment (slouken) [inline src/SDL_assert.c:109]: Can you change places like this to just `item`?

File: src/core/haiku/SDL_BeApp.cc [hunk 2]
```diff
@@ -186,7 +186,7 @@ void SDL_QuitBeApp(void)
 void SDL_BLooper::ClearID(SDL_BWin *bwin) {
     _SetSDLWindow(NULL, bwin->GetID());
     int32 i = _GetNumWindowSlots() - 1;
-    while (i >= 0 && GetSDLWindow(i) == NULL) {
+    while (i >= 0 && !GetSDLWindow(i)) {
```
Comment (slouken) [inline src/core/haiku/SDL_BeApp.cc:189]: For function calls, can you leave them the old `== NULL` style?

File: src/hidapi/libusb/hid.c [hunk 3]
```diff
@@ -716,7 +716,7 @@ static void invasive_fill_device_info_usage(struct hid_device_info *cur_dev, lib
 static struct hid_device_info * create_device_info_for_device(libusb_device *device, libusb_device_handle *handle, struct libusb_device_descriptor *desc, int config_number, int interface_num, int interface_class, int interface_subclass, int interface_protocol)
 {
 	struct hid_device_info *cur_dev = calloc(1, sizeof(struct hid_device_info));
-	if (cur_dev == NULL) {
+	if (!cur_dev) {
```
Comment (slouken) [inline src/hidapi/libusb/hid.c:719]: We shouldn't touch the third-party code. 
We can probably reuse the exclude list from check_stdlib_usage.py:
```
    excluded_paths = [
        "src/stdlib",
        "src/libm",
        "src/hidapi",
        "src/video/khronos",
        "include/SDL3",
        "build-scripts/gen_audio_resampler_filter.c",
        "build-scripts/gen_audio_channel_conversion.c" ]
```

Comment (slouken) [reply src/hidapi/libusb/hid.c:719]: Actually, we should include src/stdlib, but the others can be excluded.

**Comment by 1bsyl** [comment]
doing another attempt, here: https://github.com/libsdl-org/SDL/pull/8510

**Pull request closed without merging**